components/new-contribution-flow/StepProfileLoggedInForm.js

```diff
@@ -27,28 +27,37 @@ const NewContributionFlowStepProfileLoggedInForm = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [defaultSelectedProfile]);
 
-  // if the user doesn't have an incognito profile yet, we offer to create one
-  if (canUseIncognito) {
-    const incognitoProfile = profiles.find(p => p.type === 'USER' && p.isIncognito);
-    if (!incognitoProfile) {
-      profiles.push({
-        id: 'incognito',
-        type: 'USER',
-        isIncognito: true,
-        name: intl.formatMessage(messages['incognito']),
-      });
+  const filteredProfiles = React.useMemo(() => {
+    const filtered = profiles;
+
+    // if admin of collective you are donating to, remove it from the list
+    remove(filtered, p => p.id === collective.legacyId);
+
+    // if the user doesn't have an incognito profile yet, we offer to create one
+    if (canUseIncognito) {
+      const incognitoProfile = filtered.find(p => p.type === 'USER' && p.isIncognito);
+      if (!incognitoProfile) {
+        profiles.push({
+          id: 'incognito',
+          type: 'USER',
+          isIncognito: true,
+          name: intl.formatMessage(messages['incognito']),
+        });
+      }
+    } else {
+      remove(filtered, p => p.isIncognito);
```

Comment:
This is still mutating `profiles`, because they point to the same variable. Try this:

```es6
const list1 = [];
const list2 = list1;
list2.push(42);
console.log(list1);
```

There's also a `profiles.push` in there:

```suggestion
    const filtered = [...profiles];

    // if admin of collective you are donating to, remove it from the list
    remove(filtered, p => p.id === collective.legacyId);

    // if the user doesn't have an incognito profile yet, we offer to create one
    if (canUseIncognito) {
      const incognitoProfile = filtered.find(p => p.type === 'USER' && p.isIncognito);
      if (!incognitoProfile) {
        filtered.push({
          id: 'incognito',
          type: 'USER',
          isIncognito: true,
          name: intl.formatMessage(messages['incognito']),
        });
      }
    } else {
      remove(filtered, p => p.isIncognito);
```

Reply:
Thanks, I had missed that last `profile`